rust/src/sip/sip.rs

```diff
@@ -147,6 +151,55 @@ impl SIPState {
         }
     }
 
+    fn parse_request_tcp(&mut self, flow: *const core::Flow, stream_slice: StreamSlice) -> AppLayerResult {
+        let input = stream_slice.as_slice();
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
+            SIPFrameType::Pdu as u8,
+        );
+        SCLogDebug!("ts: pdu {:?}", _pdu);
+
+        match sip_parse_request(input) {
+            Ok((_, request)) => {
+                sip_frames_ts(flow, &stream_slice, &request);
+                self.append_request(input, request);
+            }
+            Err(Err::Incomplete(_needed)) => {
+                let consumed = input.len();
```

Comment:
Consumed should only be updated after a successful parse as we're letting the engine know it can now discard that data.

Reply:
What would you suggest to do here? Is `return AppLayerResult::ok()` enough?

Reply:
Would you please describe what kind of behaviour have you seen?

Reply:
> Would you please describe what kind of behaviour have you seen?

Nothing specific to SIP I can pinpoint, but other protocols like DNS, in particular multiple DNS requests in TCP packets.  So just wouldn't surprise me if it happened in SIP.  But its still valid for any TCP protocol anyways, so its a case we would want to support, for TCP in general.